site/app/views/grading/popup_refactor/RubricGraderView.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+/**
+ * ---------------------
+ *
+ * RubricGradearView.php
+ *
+ * This class creates the main display window for the Rubric Grader, featuring the
+ * NavigationBar and the PanelBar.
+ *
+ * The function createRubricGradeableView is called from createMainRubricGraderPage
+ * of RubricGraderController.php.
+ *
+ * ---------------------
+ */
+
+// Namespace:
+namespace app\views\grading\popup_refactor;
+
+// Includes:
+use app\views\AbstractView;
+use app\models\GradingOrder;
+use app\models\gradeable\Gradeable;
+use app\models\gradeable\GradedGradeable;
+
+// Main Class:
+class RubricGraderView extends AbstractView {
+    // ---------------------------------
+
+    // Member Variables:
+
+    /**
+     * The current gradeable being graded.
+     */
+    private $gradeable;
+
+    /**
+     * @var GradedGradeable
+     * The current submission being graded.
+     */
+    private $current_submission;
+
+    /**
+     * @var bool
+     * True if the current gradeable has peer grading.
+     */
+    private $is_peer_gradeable;
+
+    /**
+     * @var bool
+     * True if the current gradeable has teams.
+     */
+    private $is_team_gradeable;
+
+    /**
+     * @var string
+     *
+     * The access mode for the current user for this gradeable.
+     * Possible Values:
+     *  - "unblind" - Nothing about students is hidden.
+     *  - "single"  - For peer grading or for full access grading's Anonymous Mode. Graders cannot see
+     *               who they are currently grading.
+     *  - "double"  - For peer grading. In addition to blinded peer graders, students cannot
+     *               see which peer they are currently grading.
+     */
+    private $blind_access_mode = "";
+
+    // ---------------------------------
+
+
+    // ---
+
+
+    // ---------------------------------
+
+    /**
+     * Creates the Rubric Grading page visually.
+     * This function is called in reateMainRubricGraderPage of RubricGraderController.php.
+     *
+     * @param Gradeable $gradeable - The current gradeable.
+     * @param GradedGradeable $current_submission - The current submission we are looking at.
+     * @param Gradeable $gradeable - The current Gradeable.
+     * @param string $sort_type - The current way we are sorting students. Used to create the header.
+     * @param string $sort_direction -  Either "ASC" or "DESC" for ascending or descending sorting order.
+     *     Used to create the header.
+     * @param bool $is_peer_gradeable - True if the gradeable has peer grading.
+     * @param bool $is_team_gradeable - True if the gradeable is a team gradeable.
+     * @param string $blind_access_mode - Either "unblind", "single", or "double". See above for details.
+     * @param string $details_url - URL of the details page for this Gradeable.
```

Comment:
It would be good to add native PHP type annotations to all of the functions in this PR.  Given that you have provided a description of each parameter, it would be fine to have both PHPDoc annotations and native PHP type annotations.

Reply:
I'm not really sure what you mean. Aren't those annotations above?

Reply:
Oops, my bad.  I completely missed that there were already PHP type annotations.  🤦‍♂️